packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1943,6 +1944,51 @@ export class MergeTree {
         }
     }
 
+    /**
+     * Perform local rollback of previously deleted segments
+     */
+    public rollbackRemovedSegment(segments: ISegment[]) {
+        for (const segment of segments) {
+            segment.removedClientIds = undefined;
```

Comment:
maybe worth noting that this code is only valid b/c we assume there are no sequenced ops applied between when a local op and its rollback happens, and therefore `removedClientIds` could have only contained the local client since a concurrent delete is impossible.

If we're not likely to change that design ever, I dunno if this is a big deal. I was going to say it might be a helpful breadcrumb for people understanding `removedClientIds`. But we should probably just document that field better anyway--its semantics aren't very hard to understand, but inferring them from usage isn't easy.

Reply:
Added an assert which hopefully makes this clearer.